pandas/tests/groupby/test_rank.py

```diff
@@ -444,6 +444,7 @@ def test_rank_avg_even_vals():
     tm.assert_frame_equal(result, exp_df)
 
 
+@pytest.mark.xfail(reason="Works now, needs tests")
```

Comment:
To keep this PR focused on the approach to deduplication, plan to leave this for a followup (basically add whatsnew, tests, which will close #38278)